lisa/runners/lisa_runner.py

```diff
@@ -635,6 +635,40 @@ def _merge_test_requirements(
                         original_node_requirement = schema.load_by_type(
                             schema.NodeSpace, node_requirement_data
                         )
+
+                        # Manage the union of the platform requirements and the node
+                        # requirements before taking the intersection of
+                        # the rest of the requirements.
+                        platform_requirement.features = search_space.SetSpace(
```

Comment:
The platform_requirement is used for all test cases in this method. It doesn't work, if the test run contains test cases with different feature requirements. But it's a good way to fix it. So, you can call the `_create_platform_requirement` in this loop, instead of calling it out of loop.